test/e2e/dataset-delete.spec.js

```diff
@@ -490,13 +494,171 @@ describe('Dataset delete tests', () => {
 
     });
 
-    afterEach(() => {
+    describe('Delete by user', () => {
+        it('Deleting all dataset of an user while not logged should return a 401 error', async () => {            
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+
+
+            const deleteResponse = await requester
+                .delete(`/api/v1/dataset/by-user/${USERS.USER.id}`)
+                .send();
+
+            deleteResponse.status.should.equal(401);
+            deleteResponse.body.should.have.property('errors').and.be.an('array');
+            deleteResponse.body.errors[0].should.have.property('detail').and.equal(`Unauthorized`);
+        });
+
+        it('Deleting all dataset of an user as not that same user or an admin should return a 403 error', async () => {
+            mockGetUserFromToken(USERS.MANAGER);
+            
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+            await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
+
+
+            const deleteResponse = await requester
+                .delete(`/api/v1/dataset/by-user/${USERS.USER.id}`)
+                .set('Authorization', `Bearer abcd`)
+                .send();
+
+            deleteResponse.status.should.equal(403);
+            deleteResponse.body.should.have.property('errors').and.be.an('array');
+            deleteResponse.body.errors[0].should.have.property('detail').and.equal(`Forbidden`);
+        });
+
+        it('Deleting a dataset owned by the user as an ADMIN should return all deleted datasets', async () => {
+            mockGetUserFromToken(USERS.ADMIN);
+
+            nock(process.env.GATEWAY_URL)
+            .get(`/auth/user/${USERS.USER.id}`)
+            .reply(200, {
+                data: {
+                    ...USERS.USER,
+                }
+            });
+            
+            const fakeDatasetOne = await new Dataset(createDataset('cartodb', { userId: USERS.USER.id })).save();
```

Comment:
add a few datasets NOT owned by the same user, and validate that they stay in the db